src/gti.jl

```diff
@@ -1,3 +1,194 @@
+
+    if "START" in FITSIO.colnames(gtihdu)
+        startstr = "START"
+        stopstr = "STOP"
+    else
+        startstr = "Start"
+        stopstr = "Stop"
+    end
+
+    gtistart = read(gtihdu,startstr)
+    gtistop = read(gtihdu,stopstr)
+
+    return mapreduce(permutedims, vcat, 
+    [[a, b] for (a,b) in zip(gtistart, gtistop)])
+end
+
+function check_gtis(gti::AbstractMatrix)
+
+    if ndims(gti) != 2 || size(gti,2) != 2
+        throw(ArgumentError("Please check the formatting of the GTIs. 
+        They need to be provided as [[gti00 gti01]; [gti10 gti11]; ...]."))
+    end
+
+    gti_start = @view gti[:, 1]
+    gti_end = @view gti[:, 2]
+
+    if any(gti_end < gti_start)
+        throw(ArgumentError(
+            "The GTI end times must be larger than the GTI start times."
+        )) 
+    end
+
+    if any(@view(gti_start[2:end]) < @view(gti_end[1:end-1]))
+        throw(ArgumentError(
+            "This GTI has overlaps"
+        ))
+    end
+end
+
+function create_gti_mask(times::AbstractVector{<:Real},gtis::AbstractMatrix{<:Real};
+                         safe_interval::AbstractVector{<:Real}=[0,0], min_length::Real=0,
+                         dt::Real = -1, epsilon::Real = 0.001)
+
+    if length(times) == 0
+        throw(ArgumentError("Passing an empty time array to create_gti_mask"))
+    end
+
+    check_gtis(gtis)
+    mask = zeros(Bool,length(times))
+
+    if min_length>0
+        gtis = gtis[min_length .< @view(gtis[:,2]) - @view(gtis[:,1]),:]
+            
+        if size(gtis,1) < 1
+            @warn "No GTIs longer than min_length $(min_length)"
+            return mask, gtis
+        end
+    end   
+
+    if dt < 0
+        dt = Statistics.median(diff(times))
+    end
+    epsilon_times_dt = epsilon * dt
+
+    new_gtis = [[0.0, 0.0] for _ in range(1,size(gtis,1))]
+    new_gti_mask = zeros(Bool, size(gtis,1))
+
+    gti_start = @view gtis[:, 1]
+    gti_end = @view gtis[:, 2]
+
+    for (ig,(limmin,limmax)) in enumerate(zip(gti_start,gti_end))
+        limmin += safe_interval[1]
+        limmax -= safe_interval[2]
+        if limmax - limmin >= min_length
+            new_gtis[ig][:] .= limmin, limmax
+            for (i,t) in enumerate(times) 
+                if t >= (limmin + dt / 2 - epsilon_times_dt) && t <= (limmax - dt / 2 + epsilon_times_dt)
+                mask[i] = true
+                end
+            end
+            new_gti_mask[ig] = true
+        end
+    end
+
+    return mask, mapreduce(permutedims, vcat, keepat!(new_gtis,new_gti_mask))
+end
+
+function create_gti_from_condition(time::AbstractVector{<:Real}, condition::AbstractVector{Bool};
+    safe_interval::AbstractVector{<:Real}=[0,0], dt::AbstractVector{<:Real}=Float64[])
+    
+    if length(time) != length(condition)
+        throw(ArgumentError("The length of the condition and 
+        time arrays must be the same."))
+    end
+
+    idxs = contiguous_regions(condition)
+
+    if(isempty(dt))
```

Comment:
No need to use parentheses around conditionals
```suggestion
    if isempty(dt)
```